### ISSUE TITLE
Reason for using nativeTensorFlow instead of using TensorFlow-Keras

### ISSUE THREAD

**Issue by shan18** [mozilla/DeepSpeech#2368]
In DeepSpeech.py, the dense layer has been implemented from scratch instead of using `tf.layers.dense` or `tf.keras.layers.Dense`. Is there any specific reason as to why has it been implemented manually instead of using the dense function provided by TensorFlow?  
Similarly, why was LSTMBlockFusedCell used instead of `tf.keras.layers.LSTM`?

Lastly, if the model is implemented purely using tensorflow-keras API functions, will it give the same accuracy or was there any specific reason for choosing native TensorFlow?

**Comment by reuben** [comment]
The implementation predates those APIs. We haven't tested it specifically, but I don't see any reason it wouldn't work the same way.

**Comment by shan18** [comment]
Ok, thank you.
